clients/privacy-center/pages/_app.tsx

```diff
@@ -14,26 +16,91 @@ import "@fontsource/inter/500.css";
 import "@fontsource/inter/600.css";
 import "@fontsource/inter/700.css";
 
-import store, { persistor } from "~/app/store";
-import "~/config/config.css";
+import {
+  hydratePrivacyCenterEnvironment,
+  loadPrivacyCenterEnvironment,
+  PrivacyCenterEnvironment,
+} from "~/app/server-environment";
+import { makeStore } from "~/app/store";
 import theme from "~/theme";
+import Head from "next/head";
+
+interface PrivacyCenterProps {
+  serverEnvironment?: PrivacyCenterEnvironment;
+}
+
+/**
+ * Perform any server-side initialization for the app.
+ * 
+ * We use this to do things like:
+ * - read ENV vars for customization
+ * - fetch the "config.json" configuration files
+ * - etc.
+ * 
+ * DEFER: Custom apps do not support "getServerSideProps" for data fetching, etc.
+ * While this "getInitialProps" works, NextJS 13+ supports a new "app/"
+ * directory which has a cleaner abstraction for this kind of server-side
+ * component logic. We should migrate to this when we're ready!
+ * (see https://beta.nextjs.org/docs/upgrade-guide)
+ */
+export async function getInitialProps(context: AppContext): Promise<PrivacyCenterProps & AppInitialProps> {
+  // NOTE: NextJS *requires* we call this and merge the results into the output
+  // see https://nextjs.org/docs/advanced-features/custom-app#caveats
+  const ctx = await App.getInitialProps(context);
+  const isServer = (typeof window === "undefined");
+  if (!isServer) {
+    // Handle the case where this runs on the client - which we don't *really*
+    // want to happen, but since getServerSideProps isn't supported for custom
+    // Apps in NextJS 12, we have to guard against this possibility...
+    // NOTE: this *should* be impossible...
+    // eslint-disable-next-line no-console
+    console.warn("Unexpected App.getInitialProps() call from client-side code!");
```

Comment:
I think this might technically be impossible... but it's all dependent on NextJS magic that I only 70% understand about how it decides if & when to call getInitialProps 👍